demos/python_demos/im2latex_demo/im2latex_demo.py

```diff
@@ -0,0 +1,227 @@
+    ie.set_config(
+        {"PERF_COUNT": "YES" if args.perf_counts else "NO"}, args.device)
+
+    # encoder part
+    encoder_model_xml = args.m_encoder
+    encoder_model_bin = os.path.splitext(encoder_model_xml)[0] + ".bin"
+    log.info("Loading encoder files:\n\t{}\n\t{}".format(
+        encoder_model_xml, encoder_model_bin))
+    encoder = ie.read_network(encoder_model_xml, encoder_model_bin)
+
+    # check if all layers are supported
+    supported_layers = ie.query_network(encoder, args.device)
+    not_supported_layers = [
+        l for l in encoder.layers.keys() if l not in supported_layers]
+    if len(not_supported_layers) != 0:
+        log.error("Following layers are not supported by the plugin for specified device {}:\n{}".
+                  format(args.device, ', '.join(not_supported_layers)))
+
+    # decoder part:
+    dec_step_model_xml = args.m_decoder
+    dec_step_model_bin = os.path.splitext(dec_step_model_xml)[0] + ".bin"
+
+    log.info("Loading decoder files:\n\t{}\n\t{}".format(
+        dec_step_model_xml, dec_step_model_bin))
+    dec_step = ie.read_network(dec_step_model_xml, dec_step_model_bin)
+
+    # check if all layers are supported
+    supported_layers = ie.query_network(dec_step, args.device)
+    not_supported_layers = [
+        l for l in dec_step.layers.keys() if l not in supported_layers]
+    if len(not_supported_layers) != 0:
+        log.error("Following layers are not supported by the plugin for specified device {}:\n{}".
+                  format(args.device, ', '.join(not_supported_layers)))
+    _, _, height, width = encoder.input_info['imgs'].input_data.shape
+    target_shape = (height, width)
+    images_list = []
+    if os.path.isdir(args.input):
+        inputs = [os.path.join(args.input, inp)
+                  for inp in os.listdir(args.input)]
+    else:
+        inputs = [args.input]
+    log.info("Loading vocab file")
+    vocab = Vocab(args.vocab_path)
+
+    log.info("Loading and preprocessing images")
+    for filenm in tqdm(inputs):
+        image_raw = cv.imread(filenm)
+        assert image_raw is not None, "Error reading image {}".format(filenm)
+        image = preprocess_image(
+            PREPROCESSING[args.preprocessing_type], image_raw, target_shape)
+        record = dict(img_name=filenm, img=image, formula=None)
+        images_list.append(record)
+
+    log.info("Loading networks")
+    exec_net_encoder = ie.load_network(
+        network=encoder, device_name=args.device)
+    exec_net_decoder = ie.load_network(
+        network=dec_step, device_name=args.device)
+
+    log.info("Starting inference")
+
+    for rec in tqdm(images_list):
+        image = rec['img']
+
+        enc_res = exec_net_encoder.infer(
+            inputs={'imgs': np.expand_dims(image, axis=0)})
+        # get results
+        row_enc_out = enc_res['row_enc_out']
+        dec_states_h = enc_res['hidden']
+        dec_states_c = enc_res['context']
+        O_t = enc_res['init_0']
```

Comment:
The `O_t` name seems misleading (looks like `0_t`). Could you rename it to something more unambiguous?

Reply:
Renamed